docfx_yaml/extension.py

```diff
@@ -251,13 +252,18 @@ def _resolve_reference_in_module_summary(pattern, lines):
                 # Find the last component of the target. "~Queue.get" only returns <xref:get>
                 ref_name = ref_name[index:]
 
-            else:
+            elif pattern == REF_PATTERN_LAST:
```

Comment:
Consider (1) defining a function for each of these patterns and (2) calling all of the functions back to back with no for loop over patterns.

Reply:
I'm not exactly sure how I can achieve this with the given setup, I'll mark it as to-do for now and look back on this shortly when I refactor this codebase after launch.